### PR TITLE
Improve show lang tabs query

### PR DESCRIPTION
**Pull request by PululuK**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Improve show lang tabs query.
| Type?             | improvement 
| Category?         | CO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | N/A
| How to test?      | See how to test section bellow
| Possible impacts? | N/A

## HOW TO TEST

- Add new lang
- Copy the default shop lang data (or another) in the new language
image
- Make sure it is work fine. For example check you have new records in `ps_product_lang` table with `id_lang` = NEW LANG ID
- Delete the new lang created 
- Check that data with `id_lang` = NEW LANG ID has been deleted in all tables and with `_lang`.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/27515)

### CODE REVIEW

**Comment by Progi1984** [comment]
@PululuK Some news about this PR ?

File: classes/Language.php [hunk 1]
```diff
@@ -1761,4 +1758,28 @@ private function getCountries(string $locale): array
 
         return $countries;
     }
+
+    /**
+     * Get languages tables name in database
+     *
```
Comment (Progi1984) [inline classes/Language.php:1764]:
```suggestion
     *
     * @param array<int, string> $excludedTables
     * 
```

**Review by Progi1984 — COMMENTED** [review]
A small improvement and it's good

**Comment by kpodemski** [comment]
Hello @PululuK 

As a developer, I see the purpose of this PR, but QA can't test the behavior behind this code without any Issue. Could you create one?

**Comment by PululuK** [comment]
> Hello @PululuK
> 
> As a developer, I see the purpose of this PR, but QA can't test the behavior behind this code without any Issue. Could you create one?

Hi @kpodemski 

I just update how test section.

File: classes/Language.php [hunk 2]
```diff
@@ -1761,4 +1758,30 @@ private function getCountries(string $locale): array
...
+     * @param array<int, string> $excludedTables
+     * 
+     * @return array<int, string>
+     */
+    public static function getLanguagesTablesNames(array $excludedTables = []): array
```
Comment (Progi1984) [inline classes/Language.php]:
```suggestion
    private static function getLanguagesTablesNames(array $excludedTables = []): array
```

**Review by Progi1984 — COMMENTED** [review]
I finally change the scope of the static method `getLanguagesTablesNames` as it's only used in Language class.

Could you fix PHPCSFixer, too ?

Thanks :heart_eyes_cat:

**Review by Progi1984 — APPROVED** [review]

**Review by matks — APPROVED** [review]

**Comment by SD1982** [comment]
Hi @PululuK
With your PR i have a error on installation process 
image
This error disapear if i disable DEV_MODE

But there is also anoher error in the "store content" section i have this
image
And i can go to the next step the "next" button have no effect

There is also a error in console
image

**Comment by PululuK** [comment]
Hi @SD1982 

I'm not sure its related to my PR, but I will check.
Thanks

**Comment by Robin-Fischer-PS** [comment]
Hi @PululuK !

There is a conflict on your PR, could you check it please ? Thanks :)

**Comment by kpodemski** [comment]
ping @PululuK

**Comment by Hlavtox** [comment]
Hi @PululuK, I checked the code and we can speed this code a by a million percent. Right now we are selecting the old results and constructing an insanely big query. If this function is called in a shop with 10k products, it's a suicide and megabytes of text in that query.

You can just build this simple query for each table and execute it. Extremely fast and simple.
```sql
INSERT INTO `ps_category_lang` (`id_category`, `id_shop`, `id_lang`, `name`, `description`, `additional_description`, `link_rewrite`, `meta_title`, `meta_keywords`, `meta_description`)
SELECT `id_category`, `id_shop`, 5, `name`, `description`, `additional_description`, `link_rewrite`, `meta_title`, `meta_keywords`, `meta_description`
FROM `ps_category_lang`
WHERE id_lang = 1;
```

**Comment by kpodemski** [comment]
Hello @PululuK 

Since we have had no news from you for more than 20 days, I'm closing this PR. Feel free to re-open it if you find and motivation to finish it. Don't hesitate to ping me if you have any questions or doubts.

**Pull request closed without merging**